src/Grpc.AspNetCore.Server/GrpcEndpointRouteBuilderExtensions.cs

```diff
@@ -90,38 +90,12 @@ public static class GrpcEndpointRouteBuilderExtensions
             return new CompositeEndpointConventionBuilder(serviceBinder.EndpointConventionBuilders);
         }
 
-        private static Type GetServiceBaseType(Type serviceImplementation)
-        {
-            // TService is an implementation of the gRPC service. It ultimately derives from Foo.TServiceBase base class.
-            // We need to access the static BindService method on Foo which implicitly derives from Object.
-            var baseType = serviceImplementation.BaseType;
-
-            // Handle services that have multiple levels of inheritence
-            while (baseType?.BaseType?.BaseType != null)
-            {
-                baseType = baseType.BaseType;
-            }
-
-            return baseType;
-        }
-
         private static void ReflectionBind<TService>(ServiceBinderBase binder, TService service)
```

Comment:
This is no longer just reflection bind. It's using attribute first but falling back to reflection.

Reply:
Searching for an attribute is still reflection yo